src/mqtt_client/mqtt_client.rs

```diff
@@ -358,8 +386,25 @@ where
         }
     }
 
+    /// Update the settings from the network stack without any specific handling.
+    ///
+    /// # Returns
+    /// True if the settings changed. False otherwise
+    pub fn update(&mut self) -> Result<bool, minimq::Error<Stack::Error>> {
+        self.handled_update(|_, _, _| Result::<(), ()>::Ok(()))
```

Comment:
I would give the handler the task to do the settings copy. That would support more complicated use cases where the desired new settings are more easily computed from the old and not the new and save one of the two settings copying.

```suggestion
        self.handled_update(|_path, old, new| { *old = new; Result::<(), ()>::Ok(()) })
```